src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,145 @@
+package nearsoft.academy.bigdata.recommendation;
+
+
+import org.apache.mahout.cf.taste.common.TasteException;
+import org.apache.mahout.cf.taste.impl.model.file.FileDataModel;
+import org.apache.mahout.cf.taste.impl.neighborhood.ThresholdUserNeighborhood;
+import org.apache.mahout.cf.taste.impl.recommender.GenericUserBasedRecommender;
+import org.apache.mahout.cf.taste.impl.similarity.PearsonCorrelationSimilarity;
+import org.apache.mahout.cf.taste.model.DataModel;
+import org.apache.mahout.cf.taste.neighborhood.UserNeighborhood;
+import org.apache.mahout.cf.taste.recommender.RecommendedItem;
+import org.apache.mahout.cf.taste.recommender.UserBasedRecommender;
+import org.apache.mahout.cf.taste.similarity.UserSimilarity;
+
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MovieRecommender {
+    String path;
+    int repMovies,repUsers;
+    int user,product,review;
+    HashMap<String, Integer> userId = new HashMap<String, Integer>();
+    HashMap<String, Integer> products = new HashMap<String, Integer>();
+    HashMap<Integer,String> productId = new HashMap<Integer,String>();
+
+    public MovieRecommender(String pathUrl) throws IOException {
+        this.path = pathUrl;
+        convert();
+    }
+
+    public void convert() throws IOException {
+        this.user=0;
```

Comment:
Initialization should be part of constructor. Also, default values of int attributes is zero